src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorLanguageServerClient.ts

```diff
@@ -120,19 +116,22 @@ export class RazorLanguageServerClient implements vscode.Disposable {
                     return;
                 }
                 this.isStarted = true;
-                this.logger.logMessage('Server started and ready!');
-                this.eventBus.emit(events.ServerStart);
+                this.logger.logMessage('Server starting!');
+                for (const listener of this.onStartListeners) {
+                    await listener();
+                }
 
+                // Succesfully started, notify listeners.
+                resolve();
```

Comment:
I don't think I follow why this was moved above the `onStartedListeners`. Do those need to be `resolve`d?

Reply:
`resolve` needs to be above onStarted because now onStarted contains a call to our semanticTokensLegend, which calls `ensureStarted`, which waits for this promise to be resolved. It's the "DEADLOCK" scenario I mentioned above.

Reply:
Ahh okay, thanks for clarifying! 😄